internal/services/webpubsub/webpubsub_resource_test.go

```diff
@@ -0,0 +1,340 @@
+	id, err := parse.WebPubsubID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := client.Webpubsub.WebPubsubClient.Get(ctx, id.ResourceGroup, id.WebPubSubName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+	return utils.Bool(true), nil
+}
+
+func (r WebpubsubResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_web_pubsub" "test" {
+  name                = "acctestWebPubsub-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  sku                 = "Standard_S1"
+  capacity            = 1
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r WebpubsubResource) complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_web_pubsub" "test" {
+  name                = "acctestWebPubsub-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+
+  sku      = "Standard_S1"
+  capacity = 1
+
+  public_network_access_enabled = true
+
+  live_trace {
+    enabled = true
+    category {
+      name    = "MessagingLogs"
+      enabled = true
+    }
+  }
+
+  local_auth_enabled = true
+  aad_auth_enabled   = true
+
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r WebpubsubResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_web_pubsub" "import" {
+  name                = azurerm_web_pubsub.test.name
+  location            = azurerm_web_pubsub.test.location
+  resource_group_name = azurerm_web_pubsub.test.resource_group_name
+
+  sku      = "Standard_S1"
+  capacity = 1
+}
+`, r.basic(data))
+}
+
+func (r WebpubsubResource) standardWithCapacity(data acceptance.TestData, capacity int) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_web_pubsub" "test" {
+  name                = "acctestWebPubsub-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+
+  sku      = "Standard_S1"
+  capacity = %d
+}
+`, r.template(data), data.RandomInteger, capacity)
+}
+
+func (r WebpubsubResource) propertyUpdate(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_web_pubsub" "test" {
+  name                = "acctestWebPubsub-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+
+  sku      = "Standard_S1"
+  capacity = 1
+
+  public_network_access_enabled = false
+
+  live_trace {
+    enabled = false
+    category {
+      name    = "MessagingLogs"
+      enabled = false
+    }
+    category {
+      name    = "ConnectivityLogs"
+      enabled = true
+    }
+    category {
+      name = "HttpRequestLogs"
+    }
```

Comment:
rather than exposing categories like this, we should be making these properties and booleans - as we've seen with SignalR doing this leads to issues unfortunately

Reply:
@tombuildsstuff Are you suggesting to modify the code as below?
```
live_trace {

"enabled": bool,
"MessagingLogs_enabled" :bool,
"ConnectivityLogs_enabled": bool,
"HttpRequestLogs_enabled":bool,
}
```

Reply:
yes, i believe that is what he is getting at

Reply:
changed the code in the corresponding PR: [Web Publishing Subscription](https://github.com/hashicorp/terraform-provider-azurerm/pull/14731)